### PR TITLE
Process morphological mode in explore graph paths

### PR DESCRIPTION
**Pull request by Eradan94**
## Description

Process the morphological mode in explore graph paths.
Furthermore, some variables like tp and ep are renamed.

## Motivation and Context

Issue #80 

## How Has This Been Tested?

Tested with several graphs in French, Malagasy and Korean.
Korean : tested with a morphological graph given by professor Nam.
Malagasy : V-tous.grf
French : see attachment and Dnum-ord.grf
[tested-graphs.zip](https://github.com/UnitexGramLab/unitex-core/files/3320190/tested-graphs.zip)

## Type of files

- [ ] `bin`: Binary files
- [ ] `ci`: Continuous integration files
- [ ] `doc`: Documentation files
- [x] Plain-text source code files

## Level of change

- [ ] `break`: Breaking change
- [ ] `exp`: Experimental change
- [ ] `tmp`: Temporal change
- [ ] `major`: Major change
- [ ] `minor`: Minor change
- [ ] `sec`: Vulnerability-related change
- [x] None of the above (normal change)

## Type of change

- [ ] `deprecat`: Deprecation of a once-stable feature
- [x] `enhance`: Enhancement in existing functionality
- [ ] `fix`: Bug fix
- [ ] `feature`: New feature
- [ ] `hotfix`: Hotfix for bugs
- [ ] `refactor`: Improve coding style, comments
- [ ] `remove`: Remove a feature

## Checklist:

- [x] My code compiles
- [x] My code follows the code style of this project
- [ ] My code includes javadoc/doxygen where appropriate
- [x] My code is well factored, so that there is not repetitive code in the wild
- [x] My code does not require a change in the documentation, if so I already opened an issue to list the changes
- [x] I have read the **CONTRIBUTING** document
- [x] I have read the **Commit Message Guidelines**
- [x] I understand that all commits on my pull request will be squashed to a single good one

### CODE REVIEW

File: Fst2List.cpp [hunk 1]
```diff
@@ -1934,8 +1936,8 @@ int CFstApp::outWordsOfGraph(int depth) {
   Fst2Tag Tag;
   unichar *suffixPtr;
   unichar *wordPtr;
-  unichar *ep; //input buffer ptr
-  unichar *tp; //transducer buffer ptr
+  unichar *ibp; //input buffer ptr
```
Comment (martinec) [inline Fst2List.cpp]:
```suggestion
  unichar *inputBufferPtr  / / input buffer ptr
```

File: Fst2List.cpp [hunk 2]
```diff
@@ -1934,8 +1936,8 @@ int CFstApp::outWordsOfGraph(int depth) {
...
+  unichar *obp; //transducer buffer ptr
```
Comment (martinec) [inline Fst2List.cpp]:
```suggestion
    unichar *outputBufferPtr  // transducer buffer ptr
```

File: Fst2List.cpp [hunk 3]
```diff
@@ -269,6 +269,7 @@ class CFstApp {
   char ofExt[16];
   char ofnameOnly[512];        // output file name
   char defaultIgnoreName[512]; // input file name
+  bool modeMorph = false; //true if the current state is in morphological mode
```
Comment (martinec) [inline Fst2List.cpp]:
```suggestion
  bool inMorphoMode = false;  // true if the current state is in morphological mode
```

File: Fst2List.cpp (hunk 2 above)
Comment (martinec) [reply Fst2List.cpp]: Giving sensible names to types and variables is much better than using obscure names that you must then explain through comments.

File: Fst2List.cpp [hunk 4]
```diff
@@ -1944,6 +1946,7 @@ int CFstApp::outWordsOfGraph(int depth) {
   inBufferCnt = outBufferCnt = 0;
   inputPtrCnt = outputPtrCnt = 0;
   unichar aaBuffer_for_getLabelNumber[64];
+  bool isWord; //false if the tag content is not a word (like $< or $>)
```
Comment (martinec) [inline Fst2List.cpp]: end-of-line comments should be separated from the code by 2 spaces

```cpp
bool isWord;[SPACE][SPACE]//[SPACE]false
```

File: Fst2List.cpp [hunk 5]
```diff
@@ -772,6 +773,23 @@ class CFstApp {
     inputPtrCnt = outputPtrCnt = inBufferCnt = outBufferCnt = 0;
   }
 
+  /**
+    * Prints a single space when the morphological mode is off
+  **/
+  void outOneSpace() {
```
Comment (martinec) [inline Fst2List.cpp]: ```...Single...```

**Comment by Eradan94** [comment]
So, I have to create 2 new PR, one for the refactoring and the other for enhancement?

**Comment by martinec** [comment]
> So, I have to create 2 new PR, one for the refactoring and the other for enhancement?

You got it right!

**Pull request closed without merging**